### PR TITLE
scheduller(ticc): fix uneven write keys distribution

### LINKED ISSUE #9665: uneven write keys distribution
img_v2_e7f2a46b-d019-44fa-b8c6-f43801d38c2g

### PR DESCRIPTION
**Pull request by sdojjy**
### What problem does this PR solve?

Issue Number: close #9665

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
 `None`.
```

### CODE REVIEW

File: cdc/scheduler/internal/v3/keyspan/splitter_write_test.go [hunk 1]
```diff
@@ -214,3 +216,23 @@ func TestSplitRegionsByWrittenKeysConfig(t *testing.T) {
 	})
 	require.Empty(t, spans)
 }
+
+func TestSplitEven(t *testing.T) {
```
Comment (asddongmen) [inline cdc/scheduler/internal/v3/keyspan/splitter_write_test.go:220]:
```suggestion
func TestSplitRegionEven(t *testing.T) {
```

File: cdc/scheduler/internal/v3/keyspan/reconciler.go [hunk 2]
```diff
@@ -208,3 +213,15 @@ func (m *Reconciler) Reconcile(
 	}
 	return m.spanCache
 }
+
+func getSpansNumber(regionNum, captureNum int) int {
+	spanNum := 1
+	if regionNum > 1 {
+		// spanNumber = max(captureNum * spanNumberCoefficient, totalRegions / spanRegionLimit)
+		spanNum = captureNum * spanNumberCoefficient
```
Comment (nongfushanquan) [inline cdc/scheduler/internal/v3/keyspan/reconciler.go:221]: I suggest spanNumberCoefficient = max(3,  captureNum -1)

File: cdc/scheduler/internal/v3/keyspan/splitter_write.go [hunk 3]
```diff
@@ -74,121 +80,121 @@ func (m *writeSplitter) split(
...
 ) *splitRegionsInfo {
 	decodeKey := func(hexkey string) []byte {
 		key, _ := hex.DecodeString(hexkey)
 		return key
 	}
-	totalWriteNormalized := uint64(0)
-	totalWrite := totalWriteNormalized
+	totalWrite, totalWriteNormalized := uint64(0), uint64(0)
 	for i := range regions {
 		totalWrite += regions[i].WrittenKeys
```
Comment (nongfushanquan) [inline cdc/scheduler/internal/v3/keyspan/splitter_write.go:111]: Please add comments to explain the meaning of the regions[I].writttenkeys.

File: cdc/scheduler/internal/v3/keyspan/splitter_write.go [hunk 4]
```diff
@@ -20,32 +20,42 @@ import (
 	"github.com/pingcap/log"
 	"github.com/pingcap/tiflow/cdc/model"
 	"github.com/pingcap/tiflow/cdc/processor/tablepb"
-	"github.com/pingcap/tiflow/pkg/config"
 	"github.com/pingcap/tiflow/pkg/pdutil"
 	"go.uber.org/zap"
 )
 
 const regionWrittenKeyBase = 1
```
Comment (nongfushanquan) [inline cdc/scheduler/internal/v3/keyspan/splitter_write.go:27]: whether 1 is enough?

File: pkg/pdutil/api_client.go [hunk 5]
```diff
@@ -169,6 +169,9 @@ type RegionInfo struct {
 	StartKey    string `json:"start_key"`
 	EndKey      string `json:"end_key"`
 	WrittenKeys uint64 `json:"written_keys"`
+	// NormalizedWeight is the written keys normalized
+	// by the average region written keys of a same table.
+	NormalizedWeight uint64
```
Comment (sdojjy) [inline pkg/pdutil/api_client.go:174]: remove this field, it's not a part of response

Comment (asddongmen) [reply pkg/pdutil/api_client.go:174]: sure, I will put it another place.

**Review by CharlesCheung96 — APPROVED** [review]

**Review by nongfushanquan — APPROVED** [review]

**Pull request merged**